data_validation/__main__.py

```diff
@@ -290,6 +290,17 @@ def build_config_from_args(args: Namespace, config_manager: ConfigManager):
             comparison_fields = cli_tools.get_arg_list(
                 args.comparison_fields, default_value=[]
             )
+
+            # As per #1190, add rstrip for Teradata string comparison fields
+            if (
+                config_manager.source_client.name == "teradata"
+                or config_manager.target_client.name == "teradata"
+            ):
+
+                comparison_fields = config_manager.add_rstrip_to_comp_fields(
```

Comment:
I'm probably missing some key point but here's an ignorant question:

It feels like this detail should be hidden away inside `build_config_comparison_fields()` and not in __main__.

Is a valid alternative adding another field to the `column_config` dict like we already do for `consts.CONFIG_CAST`? And only set the new field when source or target is Teradata and type is string?

Reply:
The tough part about this was that I wanted to add the rstrip() to validations where comparison fields where explicitly passed in. When running a hash or concat validation, the comparison field automatically populates as 'hash__all' or 'concat__all' respectively. If I updated the `column_config` dict, it will add an rtrim on top of the hash/concat values which we don't want. 

I do agree with moving the code into config_manager() though - will make that change!

Reply:
As I was implementing this, I realized that this snippet does have to remain in main.py. If I move the 'add_rstrip_to_comp_fields' logic into the `build_config_comparison_fields()` method, it adds the rstrip to hash and concat values since that function is also called when building the comparison fields for hash and concat.